src/Microsoft.PowerShell.Commands.Utility/commands/utility/ShowMarkdownCommand.cs

```diff
@@ -28,46 +30,119 @@ public class ShowMarkdownCommand : PSCmdlet
+        /// </summary>
+        [Parameter(ParameterSetName = "LiteralPath",
+                   Mandatory = true, ValueFromPipeline = false, ValueFromPipelineByPropertyName = true)]
+        [Alias("PSPath", "LP")]
+        public string[] LiteralPath
+        {
+            get { return Path; }
+            set { Path = value; }
+        }
+
         /// <summary>
         /// Gets or sets the switch to view Html in default browser.
         /// </summary>
         [Parameter]
         public SwitchParameter UseBrowser { get; set; }
 
-        private SteppablePipeline stepPipe;
+        private System.Management.Automation.PowerShell _powerShell;
 
         /// <summary>
         /// Override BeginProcessing.
         /// </summary>
         protected override void BeginProcessing()
         {
-            if (!UseBrowser.IsPresent)
-            {
-                // Since UseBrowser is not bound, we use proxy to Out-Default
-                stepPipe = ScriptBlock.Create(@"Microsoft.PowerShell.Core\Out-Default @PSBoundParameters").GetSteppablePipeline(this.MyInvocation.CommandOrigin);
-                stepPipe.Begin(this);
-            }
+            _powerShell = System.Management.Automation.PowerShell.Create();
         }
 
         /// <summary>
         /// Override ProcessRecord.
         /// </summary>
         protected override void ProcessRecord()
         {
-            object inpObj = InputObject.BaseObject;
+            switch (ParameterSetName)
+            {
+                case "InputObject":
+                    if (InputObject.BaseObject is MarkdownInfo)
+                    {
+                        ProcessMarkdownInfo(InputObject.BaseObject);
+                    }
+                    else
+                    {
+                        ConvertFromMarkdown("InputObject", InputObject.BaseObject);
+                    }
+
+                    break;
+
+                case "Path":
+                case "LiteralPath":
+                    ConvertFromMarkdown(ParameterSetName, Path);
+                    break;
+
+                default:
+                    Debug.Assert(false, string.Format(CultureInfo.InvariantCulture, "Invalid parameter set name: {0}", ParameterSetName));
+                    break;
+            }
+        }
+
+        /// <summary>
+        /// Process markdown as path.
+        /// </summary>
+        /// <param name="parameter">Name of parameter to pass to `ConvertFrom-Markdown`.</param>
+        /// <param name="input">Value of parameter.</param>
+        private void ConvertFromMarkdown(string parameter, object input)
+        {
+            _powerShell.AddCommand("Microsoft.PowerShell.Utility\\ConvertFrom-Markdown").AddParameter(parameter, input);
+            if (!UseBrowser)
+            {
+                _powerShell.AddParameter("AsVT100EncodedString");
+            }
+
+            Collection<PSObject> output = _powerShell.Invoke();
```

Comment:
You can use `Invoke<MarkdownInfo>()`, so the output will be `MarkdownInfo` collection and we can avoid the type comparison in `ProcessMarkdownInfo`.